src/include/network/services/dns.rb

```diff
@@ -431,23 +409,74 @@ def StoreHnWidget(key, _event)
       nil
     end
 
-    # Init handler for DHCP_HOSTNAME.
-    # enable or disable: is DHCP available?
-    # @param [String] key	the widget receiving the event
-    # @param event	the event being handled
-    # @return nil so that the dialog loops on
-    def InitDhcpHostname(_key)
-      UI.ChangeWidget(Id("DHCP_HOSTNAME"), :Enabled, @has_dhcp)
-      if !@has_dhcp
-        UI.ReplaceWidget(Id("dh_host_text"), Label(_("No interface with dhcp")))
+    # Checks whether setting hostname via DHCP is allowed
+    def use_dhcp_hostname?
+      UI.QueryWidget(Id("DHCP_IFACES"), :Value) != "none"
+    end
+
+    NONE_LABEL = "no".freeze
+    ANY_LABEL = "any".freeze
+    NO_CHANGE_LABEL = "no_change".freeze
+
+    # Init handler for DHCP_IFACES
+    def InitDhcpIfaces(_key)
+      UI.ChangeWidget(Id("DHCP_IFACES"), :Enabled, has_dhcp?)
+
+      hostname_ifaces = LanItems.find_set_hostname_ifaces
+      selected = DNS.dhcp_hostname ? ANY_LABEL : NONE_LABEL
+      items = []
+
+      if !DNS.valid_dhcp_cfg?
+        fix_dhclient_warning(hostname_ifaces)
+
+        selected = NO_CHANGE_LABEL
+        items = [Item(Id(NO_CHANGE_LABEL), "", true)]
```

Comment:
Do we really want an empty description here? Combined with the vague "indeterministic!" popup this would leave me really confused as to what will happen and what I should do.